Source/santad/SNTExecutionController.m

```diff
@@ -109,25 +113,46 @@ - (void)validateBinaryWithMessage:(santa_message_t)message {
   // Ignore codesigning if there are any errors with the signature. 
   if (csError) csInfo = nil;
 
-  // Actually make the decision.
+  // Actually make the decision (and refresh rule access timestamp).
   SNTCachedDecision *cd = [self.policyProcessor decisionForFileInfo:binInfo
                                                          fileSHA256:nil
                                                   certificateSHA256:csInfo.leafCertificate.SHA256];
   cd.certCommonName = csInfo.leafCertificate.commonName;
   cd.vnodeId = message.vnode_id;
 
-  // Formulate an action from the decision
+  // Formulate an initial action from the decision.
   santa_action_t action =
       (SNTEventStateAllow & cd.decision) ? ACTION_RESPOND_ALLOW : ACTION_RESPOND_DENY;
 
-  // Save decision details for logging the execution later.
-  if (action == ACTION_RESPOND_ALLOW) [_eventLog saveDecisionDetails:cd];
+  // Possibly modify that action.
+  if (cd.decision == SNTEventStateAllowCompiler &&
+      [[SNTConfigurator configurator] transitiveWhitelistingEnabled]) {
+    // If rule indicated that the allowed binary was a compiler and Santa is configured to allow
+    // transitive whitelisting, upgrade the action to ACTION_RESPOND_ALLOW_COMPILER.
```

Comment:
Could SNTPolicyProcessor not make this determination?

Reply:
Moved the check for transitiveWhitelistingEnabled to SNTPolicyProcessor so that if it's not enabled, the decision passed to SNTExecutionController from a SNTRuleStateWhitelistCompiler rule will be downgraded to SNTEventStateAllow.